packages/docs/docs/guides/5-request-client/2-erc20-payment-detection.ts

```diff
@@ -0,0 +1,211 @@
+/**
+ * # Creating a Request using an erc20 payment network
+ *
+ * This is an example of creating a request, using an erc20 payment network
+ * The request will be stored in memory and cleared as soon as the script is finished running.
+ */
+
+/**
+ * ## Basics
+ * 
+ * For both erc20 payment networks, Request Client must be initialized.
+ */
+
+/**
+ * ### Imports
+ *
+ * First we import the 2 packages we will need to create the request:
+ */
+// The signature provider allow us to sign the request
+import { EthereumPrivateKeySignatureProvider } from '@requestnetwork/epk-signature';
+// RequestNetwork is the interface we will use to interact with the Request network
+import * as RequestNetwork from '@requestnetwork/request-client.js';
+
+/**
+ * ### Identity
+ *
+ * To create a request we need to declare the identities of the parties involved.
+ * Identities are the unique identifier of a request user. They are not payment addresses, only unique addresses that identify a person/entity.
+ */
+
+// Here we declare the payee identity,  with the payee identity ethereum address
+const payeeIdentity = {
+  type: RequestNetwork.Types.Identity.TYPE.ETHEREUM_ADDRESS,
+  value: '0x627306090abab3a6e1400e9345bc60c78a8bef57',
+};
+
+// Here we declare the (optional, but recommended) payer identity address.
+const payerIdentity = {
+  type: RequestNetwork.Types.Identity.TYPE.ETHEREUM_ADDRESS,
+  value: '0xF317BedAA5c389F2C6f469FcF25e0752C7228Ba6',
+};
+
+/**
+ * ### Signature Provider
+ *
+ * The Ethereum private key signature provider allows a user to pass in their private Ethereum key to sign a request.
+ * The signature is a proof of who created the request and of it's integrity (that no data changed after it was signed).
+ * This process is similar to the signature of an Ethereum transaction.
+ */
+
+// The signature info requires the request creator private key.
+// Please be careful with how you store and handle your private key since it's a very sensitive piece of data.
+const payeeSignatureInfo = {
+  method: RequestNetwork.Types.Signature.METHOD.ECDSA,
+  privateKey: '0xc87509a1c067bbde78beb793e6fa76530b6382a4c0241e5e4a9ec0a0f44dc0d3',
+};
+
+const signatureProvider = new EthereumPrivateKeySignatureProvider(payeeSignatureInfo);
+
+// Add the payer private key to the signature provider to be able to accept the request
+const payerSignatureInfo = {
```

Comment:
this is confusing. as a doc reader, I would not get why there are private keys for both payer and payee. 

We should move it closer to the `accept` part. 

Overall comment, try to limit the number of variable declaration, they add confusion imo

Reply:
The problem is this is the furthest I can put this line without having to redeclare `requestNetwork` (because `requestNetwork` takes the `signatureProvider` in constructor argument)

Based on your last statement, I remove `payerSignatureInfo` and `accept` function since this is not really needed, it is not the aim of this page